feat: add error handling page for Storage & Auth (#8288)
aws-amplify/docs: src/pages/[platform]/build-a-backend/storage/error-handling/index.mdx

import { getCustomStaticPath } from '@/utils/getCustomStaticPath';

export const meta = {
  title: 'Error handling',
  description:
    'Handling errors from Amplify Storage',
  platforms: [
    'angular',
    'javascript',
    'nextjs',
    'react',
    'vue',
    'react-native'
  ]
};

export const getStaticPaths = async () => {
  return getCustomStaticPath(meta.platforms);
};

export function getStaticProps(context) {
  return {
    props: {
      platform: context.params.platform,
      meta
    }
  };
}

> The runtime errors thrown by the Amplify Storage APIs may contains additional information to help root-causing the problems. These information are available if the error is an instance of `StorageError`.

Reviewer: ```suggestion
The runtime errors thrown by the Amplify Storage APIs may contain additional information helpful with identifying root-causes of problems. This information is available if the error is an instance of `StorageError`.
```

```javascript
import { StorageError, uploadData } from 'aws-amplify/storage';

try {
  const result = await uploadData({
    path: "album/2024/1.jpg",
    data: file,
  }).result;
  console.log('Succeeded: ', result);
} catch (error) {
  if (error instanceof StorageError) {
    console.error('Storage error', error.name);
    if (error.metadata) {
      console.error('request ID: ', error.metadata.requestId);
    }
  } 
  // Further error handling...
}
```

> `AuthError` may also be thrown by the Storage APIs.

Reviewer 2: Maybe add a bit more detail here. Feels weird to say this and then show a table of the StorageError properties.

## All `StorageError` properties

Property | Type | Required | Description |
| -- | -- | -- | ----------- |
| name | String | Required | Client-side error name or server-side error code. |
| message | String | Required | Error message. |
| stack | String | Optional | Stack trace. |
| recoverySuggestion | String | Optional | Common cause of the error and human-readable recommendation to fix the error.|
| metadata | Object | Optional | Bag of additional error information. |
| metadata.httpStatusCode | number | Optional | The response HTTP status code if the error is caused by erroneous HTTP response. |

> | metadata.requestId | String | Optional | The request ID useful to [contact AWS Support](https://docs.aws.amazon.com/AmazonS3/latest/API/get-request-ids.html). |
> | metadata.extendedRequestId | String | Optional | The request ID useful to [contact AWS Support](https://docs.aws.amazon.com/AmazonS3/latest/API/get-request-ids.html). |

Reviewer: ```suggestion
| metadata.requestId | String | Optional | The request ID used by AWS when [contacting AWS Support](https://docs.aws.amazon.com/AmazonS3/latest/API/get-request-ids.html). |
| metadata.extendedRequestId | String | Optional | The extended request ID used by AWS when [contacting AWS Support](https://docs.aws.amazon.com/AmazonS3/latest/API/get-request-ids.html). |
```
